src/rest/server_service/path_predicate.rs

```diff
@@ -0,0 +1,26 @@
+use actix_web::pred::Predicate;
+use actix_web::Request;
+use regex::{self, Regex};
+
+pub struct PathPredicate {
+    regex: Regex,
+}
+
+impl PathPredicate {
+    pub fn new(prefix: &str, pattern: &str) -> Self {
```

Comment:
It seems that all uses of this predicate try to parse three components: the path prefix, the path segment without slashes in the middle, and the possibly empty expected path selector at the end. If it's the only thing this predicate is expected to do, I would just have the prefix and the end selector as parameters and not require the type user to supply regex syntax.